server/service/src/requisition/indicator_value/update.rs

```diff
@@ -0,0 +1,256 @@
+    RepositoryError, StorageConnection,
+};
+
+use crate::service_provider::ServiceContext;
+
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct UpdateIndicatorValue {
+    pub id: String,
+    pub value: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum UpdateIndicatorValueError {
+    DatabaseError(RepositoryError),
+    IndicatorValueDoesNotExist,
+    NotThisStoreValue,
+    ValueNotCorrectType,
+    IndicatorLineDoesNotExist,
+    IndicatorColumnDoesNotExist,
+}
+
+type OutError = UpdateIndicatorValueError;
+
+pub fn update_indicator_value(
+    ctx: &ServiceContext,
+    input: UpdateIndicatorValue,
+) -> Result<IndicatorValueRow, OutError> {
+    let indicator_value = ctx
+        .connection
+        .transaction_sync(|connection| {
+            let indicator_value_row = validate(connection, &input, ctx.store_id.clone())?;
+
+            let updated_row = generate(indicator_value_row, input);
+
+            IndicatorValueRowRepository::new(connection).upsert_one(&updated_row)?;
+
+            IndicatorValueRepository::new(connection)
+                .query_one(IndicatorValueFilter::new().id(EqualFilter::equal_to(&updated_row.id)))
+                .map_err(OutError::DatabaseError)?
+                .ok_or(OutError::IndicatorValueDoesNotExist)
+        })
+        .map_err(|error| error.to_inner_error())?;
+    Ok(indicator_value)
+}
+
+fn validate(
+    connection: &StorageConnection,
+    input: &UpdateIndicatorValue,
+    store_id: String,
+) -> Result<IndicatorValueRow, OutError> {
+    let indicator_value_row = check_indicator_value_exists(connection, &input.id)?
+        .ok_or(OutError::IndicatorValueDoesNotExist)?;
+
+    if store_id != indicator_value_row.store_id {
+        return Err(OutError::NotThisStoreValue);
+    }
+
+    let indicator_line = IndicatorLineRepository::new(connection)
+        .query_by_filter(IndicatorLineFilter::new().id(EqualFilter::equal_to(
+            &indicator_value_row.indicator_line_id,
+        )))?
+        .pop()
+        .ok_or(OutError::IndicatorLineDoesNotExist)?;
+
+    let indicator_column = IndicatorColumnRepository::new(connection)
+        .query_by_filter(IndicatorColumnFilter::new().id(EqualFilter::equal_to(
+            &indicator_value_row.indicator_column_id,
+        )))?
+        .pop()
+        .ok_or(OutError::IndicatorColumnDoesNotExist)?;
+
+    if let Some(column_value_type) = indicator_column.value_type {
+        if column_value_type == IndicatorValueType::Number {
+            match input.value.parse::<f64>() {
+                Ok(_) => {}
+                Err(_) => return Err(OutError::ValueNotCorrectType),
+            }
+        }
+    } else if let Some(line_value_type) = indicator_line.value_type {
+        if line_value_type == IndicatorValueType::Number {
+            match input.value.parse::<f64>() {
+                Ok(_) => {}
+                Err(_) => return Err(OutError::ValueNotCorrectType),
+            }
+        }
+    }
```

Comment:
Values are all in string, but want to do a strict check on the Number types (:

Reply:
And so help us god if there isn't a value type on either 😁

Reply:
Is that an actual use case that could happen!?